src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,462 @@
+            files.Add(UnconfiguredProject.MakeRooted(fileName));
+            return files;
+        }
+
+        /// <summary>
+        /// ApplyAsync is called on the UI thread and its job is to update AppliedValue to be correct based on the changes that have come through data flow after being processed
+        /// </summary>
+        protected override async Task ApplyAsync(DesignTimeInputsDelta value)
+        {
+            // Not using use the ThreadingService property because unit tests
+            _unconfiguredProjectServices.ThreadingService.VerifyOnUIThread();
+
+            DesignTimeInputsItem previousValue = AppliedValue.Value;
+
+            // Calculate the new value
+            ImmutableHashSet<string> newDesignTimeInputs;
+            ImmutableHashSet<string> newSharedDesignTimeInputs;
+            var addedDesignTimeInputs = new List<string>();
+            var removedDesignTimeInputs = new List<string>();
+            bool hasRemovedDesignTimeSharedInputs = false;
+            bool hasAddedDesignTimeSharedInputs = false;
+
+            if (value.HasFileChanges)
+            {
+                var designTimeInputs = previousValue.Inputs.ToBuilder();
+                var designTimeSharedInputs = previousValue.SharedInputs.ToBuilder();
+
+                foreach (string item in value.AddedItems)
+                {
+                    // If a property changes we might be asked to add items that already exist, so we just ignore them as we're happy using our existing CancellationSeries
+                    if (designTimeInputs.Add(item))
+                    {
+                        addedDesignTimeInputs.Add(item);
+                    }
+                }
+
+                foreach (string item in value.AddedSharedItems)
+                {
+                    if (designTimeSharedInputs.Add(item))
+                    {
+                        hasAddedDesignTimeSharedInputs = true;
+                    }
+                }
+
+                foreach (string item in value.RemovedItems)
+                {
+                    if (designTimeInputs.Remove(item))
+                    {
+                        removedDesignTimeInputs.Add(item);
+                    }
+                    if (designTimeSharedInputs.Remove(item))
+                    {
+                        hasRemovedDesignTimeSharedInputs = true;
+                    }
+                }
+
+                newDesignTimeInputs = designTimeInputs.ToImmutable();
+                newSharedDesignTimeInputs = designTimeSharedInputs.ToImmutable();
+            }
+            else
+            {
+                // If there haven't been file changes we can just flow our previous collections to the new version to avoid roundtriping our collections to builders and back
+                // Normally ShouldValueBeApplied can be used for this but our project properties require comparison to previous values so we have to apply every update
```

Comment:
Can you expand or reword this? I'm not sure what "project properties require comparison to previous values" means.

Reply:
This has been resolved by changes due to the suggestion ProjectRuleSource hint in https://github.com/dotnet/project-system/pull/4349#discussion_r238940967